kolibri/plugins/learn/assets/src/actions.js

```diff
@@ -36,7 +36,35 @@ const fetchNodes = ({ dispatch }, id) => {
   });
 };
 
+
+const searchNodes = ({ dispatch }, params, page) => {
+  // Get the collection from ContentNodeResource.
+  const contentCollection = Kolibri.resources.ContentNodeResource.getCollection();
+  const wantedFields = ['pk', 'title', 'kind', 'instance_id', 'content_id', 'description', 'files'];
+  const pageSize = 20;
+  contentCollection.fetch({
+    search: params,
+    fields: wantedFields,
+    page_size: pageSize,
+    page,
+  }).then((data) => {
+    const nodes = data[0].results;
```

Comment:
The pagination will be playing merry havoc with the API resource layer, and I see no modifications to that code to accommodate the pagination - what would your ideal API For this look like?